src/Administration/Resources/app/administration/src/module/sw-order/view/sw-order-create-initial/sw-order-create-initial.spec.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * @package checkout
+ */
+
+import { mount } from '@vue/test-utils';
+import orderStore from 'src/module/sw-order/state/order.store';
+
+const existingCustomerId = '1234';
+const notExistingCustomerId = '9876';
+
+const responses = global.repositoryFactoryMock.responses;
+
+responses.addResponse({
+    method: 'Post',
+    url: '/search/customer',
+    status: 200,
+    response: {
+        data: [
+            {
+                id: existingCustomerId,
+                attributes: {
+                    id: existingCustomerId,
+                },
+                relationships: [],
+            },
+        ],
+    },
+});
+
+async function createWrapper(customerId = null) {
+    return mount(await wrapTestComponent('sw-order-create-initial', { sync: true }), {
+        global: {
+            mocks: {
+                $route: {
+                    query: {
+                        customerId: customerId,
+                    },
+                },
+            },
+        },
+    });
+}
+
+describe('src/module/sw-order/view/sw-order-create-initial', () => {
+    let wrapper;
+
+    beforeEach(async () => {
+        wrapper = await createWrapper();
+
+        if (Shopware.State.get('swOrder')) {
+            Shopware.State.unregisterModule('swOrder');
+        }
+
+        Shopware.State.registerModule('swOrder', orderStore);
+
+        await flushPromises();
+    });
+
+    afterEach(async () => {
+        if (wrapper) {
+            await wrapper.unmount();
+        }
+
+        await flushPromises();
+    });
+
+    it('should be a Vue.js component', async () => {
+        expect(wrapper.vm).toBeTruthy();
+    });
+
+    it('should not load a customer if no customerId query parameter has been passed', async () => {
+        // ToDo: Implement test
```

Comment:
To test this, you can override repository and state functions with `jest.fn()` and expect them not being called. For example
```
wrapper.vm.customerRepository.get = jest.fn();
[...]
expect(wrapper.vm.customerRepository.get).not.toHaveBeenCalled();
```
btw. you can override anything you need in your test, this also means that your customerId doesn't have to be a createWrapper argument, you could just `wrapper.vm.$route.query.customerId = ..` - although having it as an argument is actually more expressive here :wink:

Reply:
Perfect, thank you! That's what I did. Can you do a final check now?